Info/handler.ts

```diff
@@ -1,27 +1,37 @@
 import * as express from "express";
 import { wrapRequestHandler } from "io-functions-commons/dist/src/utils/request_middleware";
 import {
+  IResponseErrorInternal,
   IResponseSuccessJson,
+  ResponseErrorInternal,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
 import * as packageJson from "../package.json";
+import { checkApplicationHealth, HealthCheck } from "../utils/healthcheck";
 
 interface IInfo {
   version: string;
 }
 
-type InfoHandler = () => Promise<IResponseSuccessJson<IInfo>>;
+type InfoHandler = () => Promise<
+  IResponseSuccessJson<IInfo> | IResponseErrorInternal
+>;
 
-export function InfoHandler(): InfoHandler {
-  return async () => {
-    return ResponseSuccessJson({
-      version: packageJson.version
-    });
-  };
+export function InfoHandler<T>(healthCheck: HealthCheck<T>): InfoHandler {
+  return () =>
+    healthCheck
+      .fold<IResponseSuccessJson<IInfo> | IResponseErrorInternal>(
+        problems => ResponseErrorInternal(problems.join("\n")),
+        _ =>
+          ResponseSuccessJson({
```

Comment:
something like this: https://tools.ietf.org/html/draft-inadarei-api-health-check-04
(but simpler)

Reply:
> return all the statuses even in case of success

What d'you mean? We are checking only the current application status - at most, if it can reach an external resources. So I think we just have a single pass/fail status

> common vocabulary for status checks.

I agree indeed.

> something like this: https://tools.ietf.org/html/draft-inadarei-api-health-check-04

I like this approach, but I notice that the above is a draft. Meanwhile, we are using a `problem+json` which is already a standard: https://tools.ietf.org/html/rfc7807. 

As for now, we need:
* a machine-readable encoding of the service pass/fail status
   |> we use http status code for that
* a human-readable report of failures, for throubleshooting
   |> we user `detail` section of the `ProblemJson` schema for that

We can model further and define a proper encoding for failure reports, to make them machine-readable. However, I see no use for that so far.